### ISSUE TITLE
Some files are added more than once which breaks GNU tar 1.33

### ISSUE THREAD

**Issue by saraedum** [megastep/makeself#239]
The latest GNU tar (1.33) has trouble extracting archives that have the same symlink added more than once. This leads to issues in makeself. In the following example, both `subdir/` and `subdir/link` are added to the archive, which leads to `subdir/link` being added twice:

```
$ mkdir -p broken/subdir                                                                                                                                                                         
$ ln -s ../a/target broken/subdir/link
$ makeself broken broken.run label startup                                                                                                                                                        Header is 666 lines long

About to compress 0 KB of data...
Adding files to archive named "broken.run"...
./subdir/
./subdir/link
./subdir/link
CRC: 3930644181
MD5: 5f6e03bad965272faf0a78d02561925f

Self-extractable archive "broken.run" successfully created.
$ ./broken.run
Verifying archive integrity...  100%   MD5 checksums are OK. All good.
Uncompressing label  100%   ... Extraction failed.
Signal caught, cleaning up
```

Why is makeself adding the directory and the symlink here? Is there something wrong with the `find` statement that determines the files that should be added?

Btw., the underlying issue in tar 1.33 can be seen here:

```
$ mkdir -p broken/subdir
$ ln -s ../a/target broken/subdir/link
$ tar cvf broken.tar broken/subdir/link broken/subdir/link
broken/subdir/link
broken/subdir/link
$ tar vxf broken.tar
broken/subdir/link
broken/subdir/link
tar: broken/subdir/link: Cannot hard link to ‘broken/subdir/link’: No such file or directory
tar: Exiting with failure status due to previous errors
```

**Comment by megastep** [comment]
Well if this is indeed a bug in tar I'm not sure what you suggest we can do about it? In the end it is entirely responsible for extraction/creation of the underlying tar archive and there's not much control over what it does here. I think the output you're seeing with the duplicate is actually the output from tar itself.

**Comment by realtime-neil** [comment]
@saraedum The `find` invocation used by `makeself.sh` simply prints non-directory members to the series of pipes that (eventually) appends members to the embedded archive --- there's nothing complicated about that treatment. It sounds like your `tar` is doing something unpleasant with broken symlinks.

**Comment by saraedum** [comment]
Thanks for looking into this. In fact this was fixed in #223. Before the find command picked up the directory and the symlink which lead to the symlink being added twice.

Sorry for the noise. I had misread the release notes of makeself 2.4.3 and assumed that the changes there were unrelated.
